src/v2/manifest/manifest_schema1.rs

```diff
@@ -49,4 +50,27 @@ impl ManifestSchema1Signed {
             .map(|l| l.blob_sum.clone())
             .collect()
     }
+
+    /// Get a collection of all image labels stored in the history array of this manifest.
+    ///
+    /// Note that for this manifest type any `layer` beyond 0 probably returns None.
+    pub fn get_labels(&self, layer: usize) -> Option<(HashMap<String, String>)> {
+        match serde_json::from_str::<serde_json::Value>(&self.history.iter().nth(layer)?.v1_compat)
+        {
+            Err(e) => {
+                debug!("{}", e);
+                None
+            }
+            Ok(v1compat_first) => Some(
+                v1compat_first
+                    .get("config")?
+                    .get("Labels")?
+                    .as_object()?
+                    .into_iter()
+                    .filter_map(|(label, value)| {
+                        Some((label.as_str().to_owned(), value.as_str()?.to_owned()))
```

Comment:
I'm not sure about the implementation details:
From https://doc.rust-lang.org/std/borrow/trait.ToOwned.html#tymethod.to_owned:

> Creates owned data from borrowed data, usually by cloning.

This would work too, would you prefer that?
```rust
                        Some((label.as_str().into(), value.as_str()?.into()))
```

Reply:
I thought that `label` was already a `String` which only needs to be cloned. If that isn't case, I'm fine with the current form. What's the type of `label`?

Reply:
`label` is of type `&std::string::String`. I've changed it so it only uses `to_owned()` now which works for both. This seems to be most readable as it clearly shows the intention of passing an owned version of the data out of the method.